chainercv/experimental/links/model/fcis/fcis_resnet101.py

```diff
@@ -0,0 +1,234 @@
+            min_size=600, max_size=1000,
+            ratios=[0.5, 1, 2], anchor_scales=[8, 16, 32],
+            rpn_initialW=None,
+            loc_normalize_mean=(0.0, 0.0, 0.0, 0.0),
+            loc_normalize_std=(0.2, 0.2, 0.5, 0.5),
+            iter2=True,
+            proposal_creator_params={
+                'nms_thresh': 0.7,
+                'n_train_pre_nms': 6000,
+                'n_train_post_nms': 300,
+                'n_test_pre_nms': 6000,
+                'n_test_post_nms': 300,
+                'force_cpu_nms': False,
+                'min_size': 16
+            }
+    ):
+        if n_fg_class is None:
+            if pretrained_model not in self._models:
+                raise ValueError(
+                    'The n_fg_class needs to be supplied as an argument')
+            n_fg_class = self._models[pretrained_model]['n_fg_class']
+
+        if rpn_initialW is None:
+            rpn_initialW = chainer.initializers.Normal(0.01)
+
+        extractor = ResNet101Extractor()
+        rpn = RegionProposalNetwork(
+            1024, 512,
+            ratios=ratios,
+            anchor_scales=anchor_scales,
+            feat_stride=self.feat_stride,
+            initialW=rpn_initialW,
+            proposal_creator_params=proposal_creator_params
+        )
+        head = FCISResNet101Head(
+            n_fg_class + 1,
+            roi_size=21, group_size=7,
+            spatial_scale=1. / self.feat_stride,
+            loc_normalize_mean=loc_normalize_mean,
+            loc_normalize_std=loc_normalize_std,
+            iter2=iter2)
+
+        mean = np.array([123.15, 115.90, 103.06],
+                        dtype=np.float32)[:, None, None]
+
+        super(FCISResNet101, self).__init__(
+            extractor, rpn, head,
+            mean, min_size, max_size,
+            loc_normalize_mean, loc_normalize_std)
+
+        if pretrained_model in self._models:
+            path = download_model(self._models[pretrained_model]['url'])
+            chainer.serializers.load_npz(path, self)
+
+
+class ResNet101Extractor(chainer.Chain):
+
+    def __init__(self):
+        super(ResNet101Extractor, self).__init__()
+        initialW = chainer.initializers.HeNormal()
+        kwargs = {
+            'initialW': initialW,
+            'bn_kwargs': {'eps': 1e-5},
+            'stride_first': True
+        }
+
+        with self.init_scope():
+            # ResNet
+            self.conv1 = Conv2DBNActiv(
+                3, 64, 7, 2, 3, nobias=True, initialW=initialW)
+            self.pool1 = lambda x: F.max_pooling_2d(x, ksize=3, stride=2)
+            self.res2 = ResBlock(3, 64, 64, 256, 1, **kwargs)
+            self.res3 = ResBlock(4, 256, 128, 512, 2, **kwargs)
+            self.res4 = ResBlock(23, 512, 256, 1024, 2, **kwargs)
+            self.res5 = ResBlock(3, 1024, 512, 2048, 1, 2, **kwargs)
+
+    def __call__(self, x):
+        with chainer.using_config('train', False):
+            with chainer.function.no_backprop_mode():
```

Comment:
I prefer setting related backpropagation in the training code.
Can you remove `with chainer.function.no_backprop_mode` from here?

Reply:
Hmm, in that case, I cannot reuse this model in training code...

Reply:
So, you prefer setting `disable_update()` by hand in training code?
OK, i will do that.
In fact, I think `no_backprop_mode` and `disable_update` are confusing me many times.